src/mongo/util/net/listen.cpp

```diff
@@ -51,6 +52,23 @@
 
 #endif
 
+// Make sure our Linux has the timer we need
+#if defined(__linux__)
+    #ifndef CLOCK_MONOTONIC_COARSE
+        #error "Your ancient kernel doesn't support CLOCK_MONOTONIC_COARSE: upgrade to v2.6.31 or newer"
```

Comment:
We still support RHEL 5, which is still at 2.6.18, but with a lot of
patches.


On Wed, Jan 14, 2015 at 4:55 PM, Ben McCann <notifications@github.com>
wrote:

> In src/mongo/util/net/listen.cpp
> <https://github.com/mongodb/mongo/pull/416#discussion-diff-22973616>:
>
> > @@ -51,6 +52,23 @@
> >
> >  #endif
> >
> > +// Make sure our Linux has the timer we need
> > +#if defined(__linux__)
> > +    #ifndef CLOCK_MONOTONIC_COARSE
> > +        #error "Your ancient kernel doesn't support CLOCK_MONOTONIC_COARSE: upgrade to v2.6.31 or newer"
>
> @erh <https://github.com/erh> what kernel versions need to be supported
> today? I'd like to try to resurrect this PR, but need to know what to
> support
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mongodb/mongo/pull/416/files#r22973616>.
>